diff -ruN boost-1_57_0/libs/filesystem/src/operations.cpp boost-1_57_0-new/libs/filesystem/src/operations.cpp
--- boost-1_57_0/libs/filesystem/src/operations.cpp	2014-10-29 09:34:00.000000000 -0600
+++ boost-1_57_0-new/libs/filesystem/src/operations.cpp	2014-11-05 15:14:30.000000000 -0700
@@ -1417,7 +1417,8 @@
     //   "http://man7.org/linux/man-pages/man2/fchmodat.2.html"
 #   if defined(AT_FDCWD) && defined(AT_SYMLINK_NOFOLLOW) \
       && !(defined(__SUNPRO_CC) || defined(__sun) || defined(sun)) \
-      && !(defined(linux) || defined(__linux) || defined(__linux__))
+      && !(defined(linux) || defined(__linux) || defined(__linux__)) \
+      && !(defined(__APPLE__))
       if (::fchmodat(AT_FDCWD, p.c_str(), mode_cast(prms),
            !(prms & symlink_perms) ? 0 : AT_SYMLINK_NOFOLLOW))
 #   else  // fallback if fchmodat() not supported
diff -ruN boost-1_57_0/libs/iostreams/build/Jamfile.v2 boost-1_57_0-new/libs/iostreams/build/Jamfile.v2
--- boost-1_57_0/libs/iostreams/build/Jamfile.v2	2014-03-31 13:56:02.000000000 -0600
+++ boost-1_57_0-new/libs/iostreams/build/Jamfile.v2	2014-11-05 08:50:40.000000000 -0700
@@ -159,8 +159,9 @@
     : $(sources) 
     : <link>shared:<define>BOOST_IOSTREAMS_DYN_LINK=1 
       <define>BOOST_IOSTREAMS_USE_DEPRECATED
-      [ ac.check-library /zlib//zlib : <library>/zlib//zlib
-        <source>zlib.cpp <source>gzip.cpp ]
+      # [ ac.check-library /zlib//zlib : <library>/zlib//zlib
+      #   <source>zlib.cpp <source>gzip.cpp ]
+      #   See https://github.com/boostorg/iostreams/commit/dfb1f61c26b77556a1cca0654d08847cf87d26ae
     :
     : <link>shared:<define>BOOST_IOSTREAMS_DYN_LINK=1
     ;
