diff -ruN doxygen-1.8.13/CMakeLists.txt doxygen-1.8.13-new/CMakeLists.txt
--- doxygen-1.8.13/CMakeLists.txt	2016-11-29 01:14:16.000000000 -0700
+++ doxygen-1.8.13-new/CMakeLists.txt	2016-12-31 12:14:54.000000000 -0700
@@ -40,8 +40,8 @@
 endif()
 
 if (${CMAKE_SYSTEM} MATCHES "Darwin")
-    set(CMAKE_CXX_FLAGS "-Wno-deprecated-register -mmacosx-version-min=10.5 ${CMAKE_CXX_FLAGS}")
-    set(CMAKE_C_FLAGS "-Wno-deprecated-register -mmacosx-version-min=10.5 ${CMAKE_C_FLAGS}")
+    set(CMAKE_CXX_FLAGS "-mmacosx-version-min=10.7 ${CMAKE_CXX_FLAGS}")
+    set(CMAKE_C_FLAGS "-mmacosx-version-min=10.7 ${CMAKE_C_FLAGS}")
     find_library(CORESERVICES_LIB CoreServices)
     set(EXTRA_LIBS ${CORESERVICES_LIB})
 endif()
