diff -ruN gras-03-04/source/scoring/utils/src/GRASVScoreWriter.cc gras-03-04.new/source/scoring/utils/src/GRASVScoreWriter.cc
--- gras-03-04/source/scoring/utils/src/GRASVScoreWriter.cc	2014-12-04 04:09:38.000000000 -0700
+++ gras-03-04.new/source/scoring/utils/src/GRASVScoreWriter.cc	2016-01-06 10:31:13.000000000 -0700
@@ -124,9 +124,13 @@
        std::map<G4int, G4double*>::iterator value = score->find(idx);
               
        if(value == score->end()) {
-	  pvData[idx]=std::make_pair<G4String, G4double>(pvName, 0.);
+	  //pvData[idx]=std::make_pair<G4String, G4double>(pvName, 0.);
+          pvData[idx]=std::make_pair(pvName, 0.);
 	} else {
-	  pvData[idx]=std::make_pair<G4String, G4double>(pvName,*(value->second)/unitValue);
+           
+          //pvData[idx]=std::make_pair<G4String, G4double>(pvName,*(value->second)/unitValue);
+          pvData[idx]=std::make_pair(pvName,*(value->second)/unitValue);
+
 	}
        
      }
@@ -251,9 +255,11 @@
        std::map<G4int, G4double*>::iterator value = score->find(idx);
               
        if(value == score->end()) {
-	  pvData[idx]=std::make_pair<G4String, G4double>(pvName, 0.);
+	  //pvData[idx]=std::make_pair<G4String, G4double>(pvName, 0.);
+          pvData[idx]=std::make_pair(pvName, 0.);
 	} else {
-	  pvData[idx]=std::make_pair<G4String, G4double>(pvName,*(value->second)/unitValue);
+	  //pvData[idx]=std::make_pair<G4String, G4double>(pvName,*(value->second)/unitValue);
+          pvData[idx]=std::make_pair(pvName,*(value->second)/unitValue);
 	}
        
      }

