diff -ruN libssh-0.6.4/cmake/Modules/DefineCompilerFlags.cmake libssh-0.6.4-new/cmake/Modules/DefineCompilerFlags.cmake
--- libssh-0.6.4/cmake/Modules/DefineCompilerFlags.cmake	2014-09-15 12:54:34.000000000 -0600
+++ libssh-0.6.4-new/cmake/Modules/DefineCompilerFlags.cmake	2015-10-06 11:40:37.639337417 -0600
@@ -10,7 +10,7 @@
     if (${CMAKE_C_COMPILER_ID} MATCHES "(GNU|Clang)")
 
         # add -Wconversion ?
-        set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -std=gnu99 -pedantic -pedantic-errors")
+        set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -std=gnu99 -pedantic")
         set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -Wall -Wextra -Wshadow -Wmissing-prototypes -Wdeclaration-after-statement")
         set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -Wunused -Wfloat-equal -Wpointer-arith -Wwrite-strings -Wformat-security")
         set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -Wmissing-format-attribute")
diff -ruN libssh-0.6.4/ConfigureChecks.cmake libssh-0.6.4-new/ConfigureChecks.cmake
--- libssh-0.6.4/ConfigureChecks.cmake	2014-12-19 01:11:17.000000000 -0700
+++ libssh-0.6.4-new/ConfigureChecks.cmake	2015-10-06 11:41:21.058374842 -0600
@@ -102,6 +102,9 @@
 check_function_exists(poll HAVE_POLL)
 check_function_exists(select HAVE_SELECT)
 check_function_exists(getaddrinfo HAVE_GETADDRINFO)
+if (WIN32)
+  set(CMAKE_EXTRA_INCLUDE_FILES winsock2.h)
+endif ()
 check_function_exists(ntohll HAVE_NTOHLL)
 check_function_exists(htonll HAVE_HTONLL)
 
@@ -116,10 +119,13 @@
     if (HAVE_WSPIAPI_H OR HAVE_WS2TCPIP_H)
         set(HAVE_GETADDRINFO TRUE)
         set(HAVE_GETHOSTBYNAME TRUE)
-        if (MSVC)
+# check_function_exists not working above.  Believe these exist for
+# VS10 (16.0+) and greater.  Known to exist for VS12 (18.0+)
+        if (NOT (${CMAKE_C_COMPILER_VERSION} VERSION_LESS 16.0))
+            message(STATUS "Overriding HAVE_NTOHLL and HAVE_HTONLL.")
             set(HAVE_NTOHLL TRUE)
             set(HAVE_HTONLL TRUE)
-        endif (MSVC)
+        endif ()
     endif (HAVE_WSPIAPI_H OR HAVE_WS2TCPIP_H)
 
     set(HAVE_SELECT TRUE)
diff -ruN libssh-0.6.4/src/connect.c libssh-0.6.4-new/src/connect.c
--- libssh-0.6.4/src/connect.c	2014-12-05 02:42:06.000000000 -0700
+++ libssh-0.6.4-new/src/connect.c	2015-10-06 11:41:21.058374842 -0600
@@ -64,6 +64,10 @@
 #include <wspiapi.h>
 #endif
 
+#ifndef EINPROGRESS
+#define EINPROGRESS WSAEINPROGRESS
+#endif
+
 #else /* _WIN32 */
 
 #include <netdb.h>
