diff -ruN libssh-0.6.4/ConfigureChecks.cmake libssh-0.6.4-new/ConfigureChecks.cmake
--- libssh-0.6.4/ConfigureChecks.cmake	2014-12-19 01:11:17.000000000 -0700
+++ libssh-0.6.4-new/ConfigureChecks.cmake	2015-02-03 19:05:06.597698000 -0700
@@ -116,10 +116,11 @@
     if (HAVE_WSPIAPI_H OR HAVE_WS2TCPIP_H)
         set(HAVE_GETADDRINFO TRUE)
         set(HAVE_GETHOSTBYNAME TRUE)
-        if (MSVC)
+# Not sure when this should be true, but certainly after VS9
+        if (_MSC_VER GREATER 1500)
             set(HAVE_NTOHLL TRUE)
             set(HAVE_HTONLL TRUE)
-        endif (MSVC)
+        endif ()
     endif (HAVE_WSPIAPI_H OR HAVE_WS2TCPIP_H)
 
     set(HAVE_SELECT TRUE)
diff -ruN libssh-0.6.4/src/connect.c libssh-0.6.4-new/src/connect.c
--- libssh-0.6.4/src/connect.c	2014-12-05 02:42:06.000000000 -0700
+++ libssh-0.6.4-new/src/connect.c	2015-02-03 19:05:13.217698000 -0700
@@ -64,6 +64,10 @@
 #include <wspiapi.h>
 #endif
 
+#ifndef EINPROGRESS
+#define EINPROGRESS WSAEINPROGRESS
+#endif
+
 #else /* _WIN32 */
 
 #include <netdb.h>
