diff -ruN matplotlib-2.0.0/lib/matplotlib/texmanager.py matplotlib-2.0.0-new/lib/matplotlib/texmanager.py
--- matplotlib-2.0.0/lib/matplotlib/texmanager.py	2017-01-15 11:16:27.000000000 -0700
+++ matplotlib-2.0.0-new/lib/matplotlib/texmanager.py	2017-04-16 07:52:58.576402700 -0600
@@ -218,8 +218,7 @@
         while '\\usepackage{type1cm}' in cmd:
             cmd.remove('\\usepackage{type1cm}')
         cmd = '\n'.join(cmd)
-        self._font_preamble = '\n'.join(['\\usepackage{type1cm}', cmd,
-                                         '\\usepackage{textcomp}'])
+        self._font_preamble = '\n'.join(['\\usepackage{type1cm}', cmd])
 
     def get_basefile(self, tex, fontsize, dpi=None):
         """
diff -ruN matplotlib-2.0.0/setupext.py matplotlib-2.0.0-new/setupext.py
--- matplotlib-2.0.0/setupext.py	2017-01-15 11:16:27.000000000 -0700
+++ matplotlib-2.0.0-new/setupext.py	2017-04-16 07:53:58.355985900 -0600
@@ -959,7 +959,12 @@
             return "Using local version for testing"
 
         if sys.platform == 'win32':
-            check_include_file(get_include_dirs(), 'ft2build.h', 'freetype')
+            incdirs = get_include_dirs()
+            for d in get_base_dirs():
+              incdirs.append(os.path.join(d, 'include\\freetype2'))
+            print_message("Potential incdirs are %s" %
+              ', '.join(`x` for x in incdirs))
+            check_include_file(incdirs, 'ft2build.h', 'freetype')
             return 'Using unknown version found on system.'
 
         status, output = getstatusoutput("freetype-config --ftversion")
@@ -973,11 +978,17 @@
         if version is None or 'No such file or directory\ngrep:' in version:
             version = self.version_from_header()
 
+        try:
+          return self._check_for_pkg_config(
+            'freetype', 'ft2build.h',
+            min_version='2.3', version=version)
+        except: pass
+
         # pkg_config returns the libtool version rather than the
         # freetype version so we need to explicitly pass the version
         # to _check_for_pkg_config
         return self._check_for_pkg_config(
-            'freetype2', 'ft2build.h',
+            'freetype', 'freetype2/ft2build.h',
             min_version='2.3', version=version)
 
     def version_from_header(self):
@@ -1021,7 +1032,7 @@
                     'lib/freetype2/include/freetype2'],
                 default_library_dirs=[
                     'freetype2/lib'],
-                default_libraries=['freetype', 'z'])
+                default_libraries=['freetype', 'png', 'z'])
             ext.define_macros.append(('FREETYPE_BUILD_TYPE', 'system'))
 
     def do_custom_build(self):
