diff -ruN netcdf-4.3.2/CMakeLists.txt netcdf-4.3.2-new/CMakeLists.txt
--- netcdf-4.3.2/CMakeLists.txt	2014-04-23 15:35:20.000000000 -0600
+++ netcdf-4.3.2-new/CMakeLists.txt	2014-05-17 09:42:32.417721067 -0600
@@ -596,7 +596,7 @@
   # .cmake files.  If this is the case, they should
   # specify HDF5_HL_LIB, HDF5_LIB, HDF5_INCLUDE_DIR manually.
   IF(HDF5_LIB AND HDF5_HL_LIB AND HDF5_INCLUDE_DIR)
-    SET(HDF5_LIBRARIES ${HDF5_LIB} ${HDF5_HL_LIB})
+    SET(HDF5_LIBRARIES ${HDF5_HL_LIB} ${HDF5_LIB})
     SET(HDF5_INCLUDE_DIRS ${HDF5_INCLUDE_DIR})
     SET(HDF5_C_LIBRARY ${HDF5_LIB})
     SET(HDF5_HL_LIBRARY ${HDF5_HL_LIB})
diff -ruN netcdf-4.3.2/liblib/CMakeLists.txt netcdf-4.3.2-new/liblib/CMakeLists.txt
--- netcdf-4.3.2/liblib/CMakeLists.txt	2014-04-23 15:35:20.000000000 -0600
+++ netcdf-4.3.2-new/liblib/CMakeLists.txt	2014-05-17 09:43:42.648661074 -0600
@@ -51,13 +51,14 @@
   IF(NOT MSVC)
     # Some version of cmake define HDF5_hdf5_LIBRARY instead of
     # HDF5_LIBRARY. Same with HDF5_HL_LIBRARY
-    IF(HDF5_hdf5_LIBRARY AND NOT HDF5_C_LIBRARY)
-      SET(HDF5_C_LIBRARY ${HDF5_hdf5_LIBRARY})
-    ENDIF()
     IF(HDF5_hdf5_hl_LIBRARY AND NOT HDF5_HL_LIBRARY)
       SET(HDF5_HL_LIBRARY ${HDF5_hdf5_hl_LIBRARY})
     ENDIF()
-    SET(TLL_LIBS ${HDF5_C_LIBRARY} ${HDF5_HL_LIBRARY} ${TLL_LIBS} ${SZIP_LIBRARY})
+    IF(HDF5_hdf5_LIBRARY AND NOT HDF5_C_LIBRARY)
+      SET(HDF5_C_LIBRARY ${HDF5_hdf5_LIBRARY})
+    ENDIF()
+    SET(TLL_LIBS ${HDF5_HL_LIBRARY} ${HDF5_C_LIBRARY} ${TLL_LIBS}
+      ${SZIP_LIBRARY} dl)
   ELSE() # Windows CMake defines HDF5_LIBRARIES.
     SET(TLL_LIBS ${HDF5_LIBRARIES} ${TLL_LIBS} ${SZIP_LIBRARY})
   ENDIF()
