diff --git a/CMakeLists.txt b/CMakeLists.txt
index 71d1b46..394dd01 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -381,6 +381,22 @@ if(OCE_REQUIRES_FREETYPE)
 	if(CMAKE_CONFIGURATION_TYPES OR NMAKE)
 		find_library(FREETYPE_LIBRARY_DEBUG freetyped "Location of freetype debug library")
 	endif(CMAKE_CONFIGURATION_TYPES OR NMAKE)
+# Repair
+        if (NOT FREETYPE_INCLUDE_DIRS)
+          set(FREETYPE_INCLUDE_DIRS)
+          if (FREETYPE_INCLUDE_DIR_ft2build)
+            set(FREETYPE_INCLUDE_DIRS ${FREETYPE_INCLUDE_DIRS} ${FREETYPE_INCLUDE_DIR_ft2build})
+          endif ()
+          if (FREETYPE_INCLUDE_DIR_freetype2)
+            set(FREETYPE_INCLUDE_DIRS ${FREETYPE_INCLUDE_DIRS} ${FREETYPE_INCLUDE_DIR_freetype2})
+          endif ()
+        endif ()
+        if (NOT FREETYPE_LIBRARIES AND FREETYPE_LIBRARY)
+          set(FREETYPE_LIBRARIES ${FREETYPE_LIBRARY})
+        endif ()
+        if (NOT FREETYPE_INCLUDE_DIRS OR NOT FREETYPE_LIBRARIES)
+          message(FATAL_ERROR "Unable to find freetype.")
+        endif ()
 endif(OCE_REQUIRES_FREETYPE)
 
 if(OCE_VISUALISATION)
