diff -ruN superlu_dist-2.5/scimake/SciCChecks.cmake superlu_dist-2.5-new/scimake/SciCChecks.cmake
--- superlu_dist-2.5/scimake/SciCChecks.cmake	2012-08-10 19:56:32.000000000 +0000
+++ superlu_dist-2.5-new/scimake/SciCChecks.cmake	2013-11-02 16:44:41.178141134 +0000
@@ -106,7 +106,7 @@
 #
 
 # Determine the processor and sse capabilities
-if (EXISTS /proc/cpuinfo)
+if (EXISTS /proc/cpuinfo AND NOT DISABLE_CPUCHECK)
   message(STATUS "Working on LINUX")
   file(READ /proc/cpuinfo cpuinfo)
   execute_process(COMMAND cat /proc/cpuinfo
