diff -ruN tables-2.3.1/blosc/blosc.c tables-2.3.1-new/blosc/blosc.c
--- tables-2.3.1/blosc/blosc.c	2011-10-09 04:25:11.000000000 -0600
+++ tables-2.3.1-new/blosc/blosc.c	2012-07-21 10:29:22.091965400 -0600
@@ -69,7 +69,9 @@
 pthread_attr_t ct_attr;          /* creation time attributes for threads */
 #endif
 
-#if defined(_POSIX_BARRIERS) && (_POSIX_BARRIERS - 20012L) >= 0
+/* #if defined(_POSIX_BARRIERS) && (_POSIX_BARRIERS - 20012L) >= 0 */
+/* below works on benten.gat.com */
+#if defined(_POSIX_BARRIERS) && (_POSIX_BARRIERS - 200112L) > 0
 #define _POSIX_BARRIERS_MINE
 #endif
 
diff -ruN tables-2.3.1/setup.py tables-2.3.1-new/setup.py
--- tables-2.3.1/setup.py	2011-10-28 14:06:48.000000000 -0600
+++ tables-2.3.1-new/setup.py	2012-07-21 10:29:22.094965600 -0600
@@ -313,9 +313,7 @@
         }
     # Copy the next DLL's to binaries by default.
     # Update these paths for your own system!
-    dll_files = ['\\windows\\system\\zlib1.dll',
-                 '\\windows\\system\\szip.dll',
-                 ]
+    dll_files = []
     if '--debug' in sys.argv:
         _platdep['HDF5'] = ['hdf5ddll', 'hdf5ddll']
 
