Index: visit/CMake/PluginMacros.cmake
===================================================================
--- visit/CMake/PluginMacros.cmake	(revision 21259)
+++ visit/CMake/PluginMacros.cmake	(working copy)
@@ -68,26 +68,15 @@
             # not configure, so the dollar sign must be escaped in the string
             # below.  Then during install, ${BUILD_TYPE} will be expanded.
             FOREACH(target ${ARGN})
-                IF(MSVC_IDE)
-                  SET(filename "${VISIT_BINARY_DIR}/exe/\${BUILD_TYPE}/${type}/${target}.dll")
-                  INSTALL(FILES ${filename}
-                    DESTINATION ${VISIT_INSTALLED_VERSION_PLUGINS}/${type}
-                    COMPONENT RUNTIME
-                    PERMISSIONS OWNER_READ OWNER_WRITE OWNER_EXECUTE 
-                                GROUP_READ GROUP_WRITE GROUP_EXECUTE 
+                SET(filename "${VISIT_BINARY_DIR}/exe/\${BUILD_TYPE}/${type}/${target}.dll")
+                INSTALL(FILES ${filename}
+                  DESTINATION ${VISIT_INSTALLED_VERSION_PLUGINS}/${type}
+                  COMPONENT RUNTIME
+                  PERMISSIONS OWNER_READ OWNER_WRITE OWNER_EXECUTE
+                                GROUP_READ GROUP_WRITE GROUP_EXECUTE
                                 WORLD_READ             WORLD_EXECUTE
-                    CONFIGURATIONS "";None;Debug;Release;RelWithDebInfo;MinSizeRel
-                  )
-                ELSE()  # For no IDE, installed straight into exe
-                  SET(filename "${VISIT_BINARY_DIR}/exe/${target}.dll")
-                  INSTALL(FILES ${filename}
-                    DESTINATION ${VISIT_INSTALLED_VERSION_PLUGINS}/${type}
-                    COMPONENT RUNTIME
-                    PERMISSIONS OWNER_READ OWNER_WRITE OWNER_EXECUTE 
-                                GROUP_READ GROUP_WRITE GROUP_EXECUTE 
-                                WORLD_READ             WORLD_EXECUTE
-                  )
-                ENDIF()
+                  CONFIGURATIONS "";None;Debug;Release;RelWithDebInfo;MinSizeRel
+                )
             ENDFOREACH(target)
         ENDIF(NOT WIN32)
         IF(APPLE)
Index: visit/databases/CMakeLists.txt
===================================================================
--- visit/databases/CMakeLists.txt	(revision 21259)
+++ visit/databases/CMakeLists.txt	(working copy)
@@ -393,6 +393,11 @@
     ENDIF(VISIT_BUILD_MINIMAL_PLUGINS)
 ENDIF(VISIT_BUILD_ALL_PLUGINS)
 
+# Everything here needs MESA on non-Windows
+if (LINUX)
+  link_directories(${MESA_LIBRARY_DIR})
+endif ()
+
 # Add the plugin directories
 FOREACH(db ${BUILD_PLUGINS})
     ADD_SUBDIRECTORY(${db})
Index: visit/databases/SimV2/CMakeLists.txt
===================================================================
--- visit/databases/SimV2/CMakeLists.txt	(revision 21259)
+++ visit/databases/SimV2/CMakeLists.txt	(working copy)
@@ -55,6 +55,9 @@
 
 
 LINK_DIRECTORIES(${VISIT_LIBRARY_DIR} ${VTK_LIBRARY_DIRS} ${ICET_LIBRARY_DIR} )
+if (LINUX)
+  link_directories(${MESA_LIBRARY_DIR})
+endif ()
 
 ADD_LIBRARY(ISimV2Database ${LIBI_SOURCES})
 TARGET_LINK_LIBRARIES(ISimV2Database visitcommon)
Index: visit/operators/CMakeLists.txt
===================================================================
--- visit/operators/CMakeLists.txt	(revision 21259)
+++ visit/operators/CMakeLists.txt	(working copy)
@@ -171,6 +171,11 @@
   LIST(APPEND BUILD_PLUGINS ModelBasedClustering)
 ENDIF(HAVE_LIB_R)
 
+# Everything here needs MESA on non-Windows
+if (LINUX)
+  link_directories(${MESA_LIBRARY_DIR})
+endif ()
+
 # Add the plugin directories
 FOREACH(operator ${BUILD_PLUGINS})
     ADD_SUBDIRECTORY(${operator})
Index: visit/plots/CMakeLists.txt
===================================================================
--- visit/plots/CMakeLists.txt	(revision 21259)
+++ visit/plots/CMakeLists.txt	(working copy)
@@ -89,6 +89,11 @@
     ENDIF(VISIT_BUILD_MINIMAL_PLUGINS)
 ENDIF(VISIT_BUILD_ALL_PLUGINS)
 
+# Everything here needs MESA on non-Windows
+if (LINUX)
+  link_directories(${MESA_LIBRARY_DIR})
+endif ()
+
 # Add the plugin directories
 FOREACH(plot ${BUILD_PLUGINS})
     ADD_SUBDIRECTORY(${plot})
Index: visit/tools/qtssh/CMakeLists.txt
===================================================================
--- visit/tools/qtssh/CMakeLists.txt	(revision 21259)
+++ visit/tools/qtssh/CMakeLists.txt	(working copy)
@@ -209,6 +209,9 @@
 
 ADD_EXECUTABLE(qtssh ${QTSSH_SOURCES} ${PLINK_SOURCES})
 TARGET_LINK_LIBRARIES(qtssh ${RELEVANT_QT_LIBS})
+if (LINUX)
+  TARGET_LINK_LIBRARIES(qtssh dl)
+endif ()
 
 VISIT_INSTALL_TARGETS(qtssh)
 VISIT_TOOLS_ADD_FOLDER(qtssh)
