Index: visit/CMakeLists.txt
===================================================================
--- visit/CMakeLists.txt	(revision 15752)
+++ visit/CMakeLists.txt	(working copy)
@@ -1858,7 +1858,10 @@
     # create a custom target that will run our nsis scripts.
     # Make sure it won't run unless specifically called.
     # 
-    IF (${MAKENSIS_FOUND})
+    if (NOT DEFINED MAKE_NSIS_INSTALLER)
+      set(MAKE_NSIS_INSTALLER TRUE)
+    endif ()
+    IF (MAKE_NSIS_INSTALLER AND ${MAKENSIS_FOUND})
         FILE(TO_NATIVE_PATH ${VISIT_SOURCE_DIR} VSD_NATIVE)
         FILE(TO_NATIVE_PATH ${CMAKE_INSTALL_PREFIX} CIP_NATIVE)
         ADD_CUSTOM_COMMAND(OUTPUT visit${VISIT_VERSION}.exe
