Index: visit/CMake/FindHDF5.cmake
===================================================================
--- visit/CMake/FindHDF5.cmake	(revision 22315)
+++ visit/CMake/FindHDF5.cmake	(working copy)
@@ -47,9 +47,10 @@
 
 INCLUDE(${VISIT_SOURCE_DIR}/CMake/SetUpThirdParty.cmake)
 
-IF (WIN32)
+OPTION(HDF5_LIBNAMES_AFFIX_DLL "Whether HDF5 library base names end with dll" ON)
+IF (WIN32 AND HDF5_LIBNAMES_AFFIX_DLL)
  SET_UP_THIRD_PARTY(HDF5 lib include hdf5dll hdf5_hldll)
-ELSE (WIN32)
+ELSE ()
   SET_UP_THIRD_PARTY(HDF5 lib include hdf5)
-ENDIF (WIN32)
+ENDIF ()
 
