Index: visit/tools/xdmf_writer/CMakeLists.txt
===================================================================
--- visit/tools/xdmf_writer/CMakeLists.txt	(revision 20080)
+++ visit/tools/xdmf_writer/CMakeLists.txt	(working copy)
@@ -51,16 +51,16 @@
 
 #********************************* SERIAL ************************************
 ADD_EXECUTABLE(xdmf_curv3d xdmf_curv3d.c)
-TARGET_LINK_LIBRARIES(xdmf_curv3d xdmf_writer_ser) 
+TARGET_LINK_LIBRARIES(xdmf_curv3d xdmf_writer_ser m) 
 
 ADD_EXECUTABLE(xdmf_ucd3d xdmf_ucd3d.c)
-TARGET_LINK_LIBRARIES(xdmf_ucd3d xdmf_writer_ser) 
+TARGET_LINK_LIBRARIES(xdmf_ucd3d xdmf_writer_ser m) 
 
 ADD_EXECUTABLE(xdmf_curv3d_wtime xdmf_curv3d_wtime.c)
-TARGET_LINK_LIBRARIES(xdmf_curv3d_wtime xdmf_writer_ser) 
+TARGET_LINK_LIBRARIES(xdmf_curv3d_wtime xdmf_writer_ser m) 
 
 ADD_EXECUTABLE(xdmf_multi_curv3d xdmf_multi_curv3d.c)
-TARGET_LINK_LIBRARIES(xdmf_multi_curv3d xdmf_writer_ser) 
+TARGET_LINK_LIBRARIES(xdmf_multi_curv3d xdmf_writer_ser m) 
 
 ADD_LIBRARY(xdmf_writer_ser xdmf_writer.c)
 TARGET_LINK_LIBRARIES(xdmf_writer_ser ${HDF5_LIB} ${ZLIB_LIB})
Index: visit/engine/main/CMakeLists.txt
===================================================================
--- visit/engine/main/CMakeLists.txt	(revision 20080)
+++ visit/engine/main/CMakeLists.txt	(working copy)
@@ -130,6 +130,10 @@
 ${VTK_INCLUDE_DIRS}
 )
 
+IF (NOT WIN32)
+  INCLUDE_DIRECTORIES(${X11_X11_INCLUDE_PATH})
+ENDIF ()
+
 # Add link directories
 LINK_DIRECTORIES(
 ${LIBRARY_OUTPUT_DIRECTORY}
