Index: visit4ctk/CMake/FindGDAL.cmake
===================================================================
--- visit4ctk/CMake/FindGDAL.cmake	(revision 22414)
+++ visit4ctk/CMake/FindGDAL.cmake	(working copy)
@@ -63,7 +63,7 @@
             INSTALL(FILES ${GDAL_LIBRARY_DIR}/gdal${VER}.dll
                 DESTINATION ${VISIT_INSTALLED_VERSION_BIN}
                 PERMISSIONS OWNER_READ OWNER_WRITE OWNER_EXECUTE GROUP_READ GROUP_WRITE GROUP_EXECUTE WORLD_READ WORLD_EXECUTE
-                CONFIGURATIONS "";None;Debug;Release;RelWithDebInfo;MinSizeRel
+                CONFIGURATIONS None Debug Release RelWithDebInfo MinSizeRel
                 )
         ENDIF(EXISTS ${GDAL_LIBRARY_DIR}/gdal${VER}.dll)
     ENDFOREACH(VER)
Index: visit4ctk/CMake/FindPySide.cmake
===================================================================
--- visit4ctk/CMake/FindPySide.cmake	(revision 22414)
+++ visit4ctk/CMake/FindPySide.cmake	(working copy)
@@ -89,7 +89,7 @@
             PERMISSIONS OWNER_READ OWNER_WRITE OWNER_EXECUTE 
                         GROUP_READ GROUP_WRITE GROUP_EXECUTE 
                         WORLD_READ WORLD_EXECUTE
-            CONFIGURATIONS "";None;Debug;Release;RelWithDebInfo;MinSizeRel
+            CONFIGURATIONS None Debug Release RelWithDebInfo MinSizeRel
             )
 
     # On OSX patch install names for the PySide module.
Index: visit4ctk/CMake/FindSilo.cmake
===================================================================
--- visit4ctk/CMake/FindSilo.cmake	(revision 22414)
+++ visit4ctk/CMake/FindSilo.cmake	(working copy)
@@ -74,7 +74,7 @@
     INSTALL(FILES ${SILO_DIR}/lib/${VISIT_MSVC_VERSION}/silex.exe
         DESTINATION ${VISIT_INSTALLED_VERSION_BIN}
         PERMISSIONS OWNER_READ OWNER_WRITE OWNER_EXECUTE GROUP_READ GROUP_WRITE GROUP_EXECUTE WORLD_READ WORLD_EXECUTE
-        CONFIGURATIONS "";None;Debug;Release;RelWithDebInfo;MinSizeRel
+        CONFIGURATIONS None Debug Release RelWithDebInfo MinSizeRel
         )
   ENDIF()
 ELSE (WIN32)
Index: visit4ctk/CMake/PluginMacros.cmake
===================================================================
--- visit4ctk/CMake/PluginMacros.cmake	(revision 22414)
+++ visit4ctk/CMake/PluginMacros.cmake	(working copy)
@@ -60,7 +60,7 @@
                 PERMISSIONS OWNER_READ OWNER_WRITE OWNER_EXECUTE 
                             GROUP_READ GROUP_WRITE GROUP_EXECUTE 
                             WORLD_READ             WORLD_EXECUTE
-                CONFIGURATIONS "";None;Debug;Release;RelWithDebInfo;MinSizeRel
+                CONFIGURATIONS None Debug Release RelWithDebInfo MinSizeRel
             )
         ELSE(NOT WIN32)
             # ${BUILD_TYPE} refers to the configuration option chosen (Release,
@@ -75,7 +75,7 @@
                   PERMISSIONS OWNER_READ OWNER_WRITE OWNER_EXECUTE
                                 GROUP_READ GROUP_WRITE GROUP_EXECUTE
                                 WORLD_READ             WORLD_EXECUTE
-                  CONFIGURATIONS "";None;Debug;Release;RelWithDebInfo;MinSizeRel
+                  CONFIGURATIONS None Debug Release RelWithDebInfo MinSizeRel
                 )
             ENDFOREACH(target)
         ENDIF(NOT WIN32)
Index: visit4ctk/CMake/ThirdPartyInstallLibrary.cmake
===================================================================
--- visit4ctk/CMake/ThirdPartyInstallLibrary.cmake	(revision 22414)
+++ visit4ctk/CMake/ThirdPartyInstallLibrary.cmake	(working copy)
@@ -99,7 +99,7 @@
                 PERMISSIONS OWNER_READ OWNER_WRITE OWNER_EXECUTE 
                             GROUP_READ GROUP_WRITE GROUP_EXECUTE 
                             WORLD_READ WORLD_EXECUTE
-                CONFIGURATIONS "";None;Debug;Release;RelWithDebInfo;MinSizeRel
+                CONFIGURATIONS None Debug Release RelWithDebInfo MinSizeRel
                 )
             # On Windows, we also need to copy the file to the 
             # binary dir so our out of source builds can run.
@@ -115,7 +115,7 @@
                 PERMISSIONS OWNER_READ OWNER_WRITE OWNER_EXECUTE 
                             GROUP_READ GROUP_WRITE GROUP_EXECUTE 
                             WORLD_READ WORLD_EXECUTE
-                CONFIGURATIONS "";None;Debug;Release;RelWithDebInfo;MinSizeRel
+                CONFIGURATIONS None Debug Release RelWithDebInfo MinSizeRel
                 )
         ENDIF(VISIT_INSTALL_THIRD_PARTY AND EXISTS ${libNAME})
 
@@ -182,7 +182,7 @@
                                 DESTINATION ${VISIT_INSTALLED_VERSION_LIB}
                                 DIRECTORY_PERMISSIONS OWNER_READ OWNER_WRITE OWNER_EXECUTE GROUP_READ GROUP_WRITE GROUP_EXECUTE WORLD_READ WORLD_EXECUTE
                                 FILE_PERMISSIONS OWNER_READ OWNER_WRITE OWNER_EXECUTE GROUP_READ GROUP_WRITE GROUP_EXECUTE WORLD_READ WORLD_EXECUTE
-                                CONFIGURATIONS "";None;Debug;Release;RelWithDebInfo;MinSizeRel
+                                CONFIGURATIONS None Debug Release RelWithDebInfo MinSizeRel
                             )
 
                             # On Apple, we need to make the framework be executable relative
@@ -199,7 +199,7 @@
                             INSTALL(FILES ${curNAMEWithExt}
                                 DESTINATION ${VISIT_INSTALLED_VERSION_LIB}
                                 PERMISSIONS OWNER_READ OWNER_WRITE OWNER_EXECUTE GROUP_READ GROUP_WRITE GROUP_EXECUTE WORLD_READ WORLD_EXECUTE
-                                CONFIGURATIONS "";None;Debug;Release;RelWithDebInfo;MinSizeRel
+                                CONFIGURATIONS None Debug Release RelWithDebInfo MinSizeRel
                             )
 
                             # On Apple, we need to make the library be executable relative.
@@ -227,7 +227,7 @@
                         DESTINATION ${VISIT_INSTALLED_VERSION_LIB}
                         DIRECTORY_PERMISSIONS OWNER_READ OWNER_WRITE OWNER_EXECUTE GROUP_READ GROUP_WRITE GROUP_EXECUTE WORLD_READ WORLD_EXECUTE
                         FILE_PERMISSIONS OWNER_READ OWNER_WRITE OWNER_EXECUTE GROUP_READ GROUP_WRITE GROUP_EXECUTE WORLD_READ WORLD_EXECUTE
-                        CONFIGURATIONS "";None;Debug;Release;RelWithDebInfo;MinSizeRel
+                        CONFIGURATIONS None Debug Release RelWithDebInfo MinSizeRel
                         PATTERN "Qt*_debug" EXCLUDE # Exclude Qt*_debug libraries in framework.
                     )
 
@@ -246,7 +246,7 @@
                     INSTALL(FILES ${tmpLIBFILE}
                         DESTINATION ${VISIT_INSTALLED_VERSION_LIB}
                         PERMISSIONS OWNER_READ OWNER_WRITE OWNER_EXECUTE GROUP_READ GROUP_WRITE GROUP_EXECUTE WORLD_READ WORLD_EXECUTE
-                        CONFIGURATIONS "";None;Debug;Release;RelWithDebInfo;MinSizeRel
+                        CONFIGURATIONS None Debug Release RelWithDebInfo MinSizeRel
                     )
     
                     # On Apple, we need to make the library be executable relative.
@@ -270,7 +270,7 @@
             INSTALL(FILES ${tmpLIBFILE}
                 DESTINATION ${VISIT_INSTALLED_VERSION_ARCHIVES}
                 PERMISSIONS OWNER_READ OWNER_WRITE GROUP_READ GROUP_WRITE WORLD_READ
-                CONFIGURATIONS "";None;Debug;Release;RelWithDebInfo;MinSizeRel
+                CONFIGURATIONS None Debug Release RelWithDebInfo MinSizeRel
             )
 
             # TODO: We could install windows import libraries here...
@@ -292,7 +292,7 @@
                 DESTINATION ${VISIT_INSTALLED_VERSION_INCLUDE}/${lcpkg}
                 DIRECTORY_PERMISSIONS OWNER_READ OWNER_WRITE OWNER_EXECUTE GROUP_READ GROUP_WRITE GROUP_EXECUTE WORLD_READ WORLD_EXECUTE
                 FILE_PERMISSIONS OWNER_READ OWNER_WRITE OWNER_EXECUTE GROUP_READ GROUP_WRITE GROUP_EXECUTE WORLD_READ WORLD_EXECUTE
-                CONFIGURATIONS "";None;Debug;Release;RelWithDebInfo;MinSizeRel
+                CONFIGURATIONS None Debug Release RelWithDebInfo MinSizeRel
                 FILES_MATCHING 
                 PATTERN "*.h"
                 PATTERN "*.H"
Index: visit4ctk/CMakeLists.txt
===================================================================
--- visit4ctk/CMakeLists.txt	(revision 22414)
+++ visit4ctk/CMakeLists.txt	(working copy)
@@ -777,7 +777,7 @@
                     PERMISSIONS OWNER_READ OWNER_WRITE OWNER_EXECUTE
                                 GROUP_READ GROUP_WRITE GROUP_EXECUTE
                                 WORLD_READ             WORLD_EXECUTE
-                    CONFIGURATIONS "";None;Debug;Release;RelWithDebInfo;MinSizeRel
+                    CONFIGURATIONS None Debug Release RelWithDebInfo MinSizeRel
                 )
             ENDIF(NOT ${pType} STREQUAL "STATIC_LIBRARY")
         ENDFOREACH(T)
@@ -790,7 +790,7 @@
             PERMISSIONS OWNER_READ OWNER_WRITE OWNER_EXECUTE
                         GROUP_READ GROUP_WRITE GROUP_EXECUTE
                         WORLD_READ             WORLD_EXECUTE
-            CONFIGURATIONS "";None;Debug;Release;RelWithDebInfo;MinSizeRel
+            CONFIGURATIONS None Debug Release RelWithDebInfo MinSizeRel
         )
         IF(APPLE)
             FOREACH(target ${ARGN})
@@ -860,7 +860,7 @@
         DIRECTORY_PERMISSIONS OWNER_READ OWNER_WRITE OWNER_EXECUTE
                               GROUP_READ GROUP_WRITE GROUP_EXECUTE
                               WORLD_READ             WORLD_EXECUTE
-        CONFIGURATIONS "";None;Debug;Release;RelWithDebInfo;MinSizeRel
+        CONFIGURATIONS None Debug Release RelWithDebInfo MinSizeRel
         FILES_MATCHING PATTERN "*.h"
         PATTERN "*.hpp"
         PATTERN ".svn" EXCLUDE
@@ -1400,7 +1400,13 @@
     # if there truly are current definitions, and more than 1 in the list, then
     # IF(${currentDefs} MATCHES NOTFOUND) causes a configure failure.
     # Grab the first item in the list for comparison instead.
-    LIST(GET currentDefs 0 firstItem )
+    # JRC, 20131126: also possible to have empty list.
+    LIST(LENGTH currentDefs listlen)
+    if (${listlen} GREATER 0)
+      LIST(GET currentDefs 0 firstItem)
+    else ()
+      LIST(set firstItem NOTFOUND)
+    endif
     IF(${firstItem} MATCHES NOTFOUND)
         SET(allDefs ${newDefs})
     ELSE(${firstItem} MATCHES NOTFOUND)
@@ -2042,7 +2048,7 @@
         DIRECTORY_PERMISSIONS OWNER_READ OWNER_WRITE OWNER_EXECUTE
                               GROUP_READ GROUP_WRITE GROUP_EXECUTE
                               WORLD_READ             WORLD_EXECUTE
-        CONFIGURATIONS "";None;Debug;Release;RelWithDebInfo;MinSizeRel
+        CONFIGURATIONS None Debug Release RelWithDebInfo MinSizeRel
     )
     INSTALL(DIRECTORY ${VISIT_MPICH_DIR}/bin
         DESTINATION ${VISIT_INSTALLED_VERSION}
@@ -2052,7 +2058,7 @@
         DIRECTORY_PERMISSIONS OWNER_READ OWNER_WRITE OWNER_EXECUTE
                               GROUP_READ GROUP_WRITE GROUP_EXECUTE
                               WORLD_READ             WORLD_EXECUTE
-        CONFIGURATIONS "";None;Debug;Release;RelWithDebInfo;MinSizeRel
+        CONFIGURATIONS None Debug Release RelWithDebInfo MinSizeRel
     )
     INSTALL(DIRECTORY ${VISIT_MPICH_DIR}/lib
         DESTINATION ${VISIT_INSTALLED_VERSION}
@@ -2062,7 +2068,7 @@
         DIRECTORY_PERMISSIONS OWNER_READ OWNER_WRITE OWNER_EXECUTE
                               GROUP_READ GROUP_WRITE GROUP_EXECUTE
                               WORLD_READ             WORLD_EXECUTE
-        CONFIGURATIONS "";None;Debug;Release;RelWithDebInfo;MinSizeRel
+        CONFIGURATIONS None Debug Release RelWithDebInfo MinSizeRel
     )
     IF(APPLE)
         FOREACH(filename libmpich.dylib libopa.dylib libmpl.dylib libpmpich.dylib)
@@ -2294,7 +2300,7 @@
         PERMISSIONS OWNER_READ OWNER_WRITE
                     GROUP_READ GROUP_WRITE
                     WORLD_READ
-        CONFIGURATIONS "";None;Debug;Release;RelWithDebInfo;MinSizeRel
+        CONFIGURATIONS None Debug Release RelWithDebInfo MinSizeRel
 )
 
 #-----------------------------------------------------------------------------
